generator/src/ClassGenerator.ts

```diff
@@ -0,0 +1,293 @@
+import { GeneratorBase } from './GeneratorBase';
+import { Helper } from './Helper';
+import { Layout } from './interface/layout';
+import { Parameter } from './interface/parameter';
+import { Schema } from './interface/schema';
+import { MethodGenerator } from './MethodGenerator';
+
+export class ClassGenerator extends GeneratorBase {
+    private generatedLines: string[];
+    private methodGenerator: MethodGenerator;
+    private classParameters: Parameter[];
+    private importList: string[];
+    private flattenedInlineParameters: Parameter[];
+    private superClassLayout: Layout | undefined;
+
+    /**
+     * Constructor
+     * @param classSchema - schema of the class to be generated
+     * @param schema - schema list
+     */
+    constructor(public readonly classSchema: Schema, schema: Schema[]) {
+        super(schema);
+        this.importList = ['Utils', 'Serializer'];
+        this.generatedLines = [];
```

Comment:
could we void mutable instance fields that can be moved to a method?

this.generatedLines:[] could be moved to the generate() method